src/fsharp/FSharp.Core/prim-types.fsi

```diff
@@ -3161,6 +3711,32 @@ namespace Microsoft.FSharp.Core
         /// <param name="action">The action to perform during the lock.</param>
         ///
         /// <returns>The resulting value.</returns>
+        /// 
+        /// <example id="lock-example">
+        /// <code lang="fsharp">
+        /// type TestCounter () =
+        ///     let mutable count = 0
+        ///     member this.PlainOldIncrement() = count <- count + 1
+        ///     member this.ThreadSafeIncrement() = lock this (this.PlainOldIncrement)
```

Comment:
Could you adjust to a sample that uses `lock this (fun () -> ...)` rather than a lock calling another method?  I'd nearly always expect lock to be used with an explicit lambda.